app/knights.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+
+
+class Knight:
+
+    def __init__(self, knight: dict) -> None:
+        self.name = knight["name"]
+        self.power = knight["power"]
+        self.hp = knight["hp"]
+        self.protection = 0
+        self.armour = knight["armour"]
+        self.weapon = knight["weapon"]
+        self.potion = knight["potion"]
+
+    def preparation(self) -> None:
```

Comment:
It is better to use a verb instead of herund in methods naming to represent an action.